api/src/test/java/com/epam/pipeline/manager/docker/DockerRegistryNotificationTest.java

```diff
@@ -16,326 +16,398 @@
 import com.epam.pipeline.manager.user.UserManager;
 import com.epam.pipeline.security.acl.AclPermission;
-import com.epam.pipeline.util.TestUtils;
-import com.fasterxml.jackson.databind.ObjectMapper;
+import com.epam.pipeline.test.acl.AbstractAclTest;
+import org.apache.commons.lang3.tuple.ImmutablePair;
 import org.junit.Assert;
 import org.junit.Before;
 import org.junit.Test;
-import org.mockito.Mock;
-import org.mockito.Mockito;
+import org.mockito.InjectMocks;
 import org.mockito.MockitoAnnotations;
 import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.boot.test.mock.mockito.MockBean;
-import org.springframework.security.test.context.support.WithMockUser;
-import org.springframework.test.annotation.DirtiesContext;
-import org.springframework.test.context.ContextConfiguration;
-import org.springframework.transaction.annotation.Propagation;
-import org.springframework.transaction.annotation.Transactional;
 
 import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.Collections;
+import java.util.Date;
 import java.util.List;
-
+import java.util.Optional;
+
+import static com.epam.pipeline.test.creator.docker.DockerCreatorUtils.getClonedTool;
+import static com.epam.pipeline.test.creator.docker.DockerCreatorUtils.getDockerRegistry;
+import static com.epam.pipeline.test.creator.docker.DockerCreatorUtils.getTool;
+import static com.epam.pipeline.test.creator.docker.DockerCreatorUtils.getToolGroup;
+import static com.epam.pipeline.test.creator.security.SecurityCreatorUtils.getEntityVO;
+import static com.epam.pipeline.test.creator.security.SecurityCreatorUtils.getUserContext;
+import static com.epam.pipeline.util.CustomAssertions.assertThrows;
 import static org.mockito.Matchers.any;
-import static org.mockito.Matchers.anyLong;
-import static org.mockito.Matchers.anyString;
-import static org.mockito.Mockito.doNothing;
-import static org.mockito.Mockito.when;
+import static org.mockito.Matchers.eq;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
 
 
-@DirtiesContext
-@ContextConfiguration(classes = TestApplicationWithAclSecurity.class)
-public class DockerRegistryNotificationTest extends AbstractManagerTest {
+public class DockerRegistryNotificationTest extends AbstractAclTest {
 
     private static final String ADMIN = "ADMIN";
-    private static final String TEST_USER = "USER";
-    private static final String TEST_REPO = "repository";
-    private static final String TEST_REPO_WITH_EXTERNAL_PATH = "repository3";
-    private static final String EXTERNAL_REPO_PATH = "external_repository";
-    private static final String TEST_REPO_WITHOUT_WRITE_ACCESS = "repository2";
-    private static final String TEST_IMAGE = "library/image";
-    private static final String TEST_IMAGE_NEW_GROUP = "library2/image";
+    private static final String USER = "USER";
```

Comment:
Could we reuse existing `SIMPLE_USER` and `ANOTHER_SIMPLE_USER` (from `AbstractAclTest`)?

Reply:
Done!